lib/AST/ASTMangler.cpp

```diff
@@ -2649,6 +2649,10 @@ void ASTMangler::appendDependentProtocolConformance(
 void ASTMangler::appendConcreteProtocolConformance(
                                       const ProtocolConformance *conformance) {
   auto module = conformance->getDeclContext()->getParentModule();
+  if (!CurGenericSignature && conformance->getGenericSignature()) {
+    CurGenericSignature =
```

Comment:
Hmm. I think this is correct, but if we set `CurGenericSignature` conditionally like this, we should remove it as we exit the function.

Reply:
@DougGregor Sure, I have used `llvm::SaveAndRestore` to do that.